src/main/java/com/yanolja_final/domain/user/controller/EmailController.java

```diff
@@ -0,0 +1,46 @@
+package com.yanolja_final.domain.user.controller;
+
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.RequiredArgsConstructor;
+import com.yanolja_final.domain.user.dto.request.EmailRequest;
+import com.yanolja_final.domain.user.service.EmailService;
+import com.yanolja_final.global.util.ResponseDTO;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/email")
+public class EmailController {
+
+    private final EmailService emailService;
+
+    @PostMapping("/confirmation")
+    public ResponseEntity<ResponseDTO<Void>> mailConfirm(@RequestBody EmailRequest emailRequest)
+        throws Exception {
+        emailService.sendVerificationEmail(emailRequest.getEmail());
+        return ResponseEntity.ok(ResponseDTO.ok());
+    }
+
+    @GetMapping("/verify/{code}")
+    public ResponseEntity<ResponseDTO<Void>> verifyEmail(@RequestParam("email") String email,
+        @PathVariable("code") String code) {
+        boolean isVerified = emailService.verifyEmailCode(email, code);
+        if (isVerified) {
+            return ResponseEntity.ok(ResponseDTO.ok());
+        } else {
+            return ResponseEntity
+                .status(HttpStatus.BAD_REQUEST)
+                .body(ResponseDTO.errorWithMessage(HttpStatus.BAD_REQUEST, "이메일 인증 실패."));
+        }
```

Comment:
```suggestion
        if (!isVerified) {
            return ResponseEntity
                .status(HttpStatus.BAD_REQUEST)
                .body(ResponseDTO.errorWithMessage(HttpStatus.BAD_REQUEST, "이메일 인증 실패."));
        } 
        
        return ResponseEntity.ok(ResponseDTO.ok());
```

이렇게 Early Return을 사용하는건 어떠실까요? (참고: https://www.howdy-mj.me/javascript/early-return)
짧은 코드에서는 큰 차이가 없긴한데 걸러내야 할 예외가 많을때 이 Early Return이 가독성면에서 빛을 발하더라고요

Reply:
이런 로직이 controller에 들어가는 것 보단 service에서 이메일 인증이 실패했다면 예외를 날리는게 좋지 않나욥..?
controller는 값을 받아와서 전달해주는 기능만 하라는 피드백을 여러번 봐서 남겨놓습니다!

Reply:
> 이렇게 Early Return을 사용하는건 어떠실까요? (참고: https://www.howdy-mj.me/javascript/early-return) 짧은 코드에서는 큰 차이가 없긴한데 걸러내야 할 예외가 많을때 이 Early Return이 가독성면에서 빛을 발하더라고요

인증 실패시 EmailVerificationException만 던지는 것으로 수정했습니다! 예외처리가 많은 로직에서 적용해보도록 하겠습니다! 공부가 되었네요 감사합니다 ㅎㅎ😊😊

Reply:
> 이런 로직이 controller에 들어가는 것 보단 service에서 이메일 인증이 실패했다면 예외를 날리는게 좋지 않나욥..? controller는 값을 받아와서 전달해주는 기능만 하라는 피드백을 여러번 봐서 남겨놓습니다!

예외처리 부분 전부 service코드로 이동하였습니다! 앞으로도 통일성을 위해 예외는 service단에서 처리하도록 할게요! 좋은 의견 감사합니다 ㅎㅎ👍👍